include/zephyr/posix/posix_features.h

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2024 BayLibre SAS
+ * Copyright (c) 2024 Tenstorrent AI ULC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```

Comment:
Maybe can guard this file with something like:
```c
#if !defined(ZEPHYR_INCLUDE_POSIX_UNISTD_H_) && !defined(ZEPHYR_INCLUDE_POSIX_SIGNAL_H_)
#warning This is a private header file, do not include it directly.
#endif
```

Reply:
It's not intended to be private.

This will probably be renamed to `sys/features.h` later, which should match newlib / picolibc.